client/main/dataset.js

```diff
@@ -12,8 +12,38 @@ Backbone.$ = $;
 const ByteBuffer = require('bytebuffer');
 
 const DataSetModel = Backbone.Model.extend({
-
     initialize() {
+        this.initializeDataSetModel();
+    },
+    initializeDataSetModel() {
+        this._changeBundles = [];
+        this._nextChangeId = 0;
+    },
+    filtersHidden() {
+        let firstColumn = this.getColumn(0, true);
+        if (firstColumn)
+            return firstColumn.columnType === 'filter' && firstColumn.hidden;
+
+        return false;
+    },
+    filterCount() {
+        let c = 0;
+        let columns = this.attributes.columns;
+        for (let i = 0; i < columns.length; i++) {
+            let column = columns[i];
+            if (column.columnType === 'filter')
+                c += 1;
+            else
+                break;
+        }
+        return c;
```

Comment:
no because this is determining the number of filter columns and the different ColumnCount's have nothing to do with columnType. Its like saying determine the number of computed columns... you'd just have to loop through and count.

Reply:
oh yup